src/controllers/AbstractController.ts

```diff
@@ -180,13 +182,32 @@ export default abstract class AbstractController<T extends AbstractService> {
 	protected parseNumberOrThrow(n: string, errorMessage: string): number {
 		const num = Number(n);
 
-		if (!Number.isInteger(num) || num < 0) {
+		if (!this.verifyInt(num)) {
 			throw new BadRequest(errorMessage);
 		}
 
 		return num;
 	}
 
+	/**
+	 * Expected format ie: 0-999
+	 */
+	protected parseRangeOfNumbersOrThrow(n: string): number[] {
+		const splitRange = n.split('-');
+		if (splitRange.length !== 2) {
+			throw new BadRequest('Incorrect range format. Expected example: 0-999');
+		}
+
+		const min = Number(splitRange[0]);
+		const max = Number(splitRange[1]);
+
+		if (!this.verifyInt(min) || !this.verifyInt(max)) {
```

Comment:
I find this logic a little bit hard to understand.

As it will accept both `0-100` and `100-0`, I kind assumed that that "reverse range" should not work.

Reply:
i.e, 

```
if !(this.verifyUInt(min) {
  throw new BadRequest('Inputted `min` invalid UInt.');
) 

if !(this.verifyNonZeroUInt(min) {
  throw new BadRequest('Inputted `max` invalid NonZeroUInt.');
) 

if (min >= max ) {
   throw new BadRequest('Invalid range: max must be bigger than min');
}
```

Reply:
Yea this is totally correct, will add that check.

Reply:
You mean with verifying UInt and a NonZeroUInt correct?

Reply:
yeah, or something even better.

Reply:
Yea i agree, it will always for concise behavior between the min and max as well. Will put that in now.

Reply:
Okay yea i cleaned it up to handle Uint correctly and Non zero uints correctly. I think its way cleaner and concise about what the error messages will return.